src/geometry/PolySetBuilder.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+
+#include "Geometry.h"
+#include "linalg.h"
+#include "GeometryUtils.h"
+#include "Polygon2d.h"
+#include "boost-utils.h"
+
+#include <vector>
+#include <string>
+#include <unordered_map>
+#include <hash.h>
+class PolySet;
+
+class PolySetBuilder
+{
+  PolySet *ps;
```

Comment:
This is fine for now, but I believe I've seen places where PolySet where managed as objects, not pointers. If we need non-pointers, we may want to just return a PolySet by value in `result()` and std::move() the arrays into the returned object.